repose-aggregator/core/repose-core/src/main/java/org/openrepose/powerfilter/PowerFilter.java

```diff
@@ -385,42 +387,25 @@ public void doFilter(ServletRequest request, ServletResponse response, FilterCha
         // Re-wrapping the request to reset the inputStream/Reader flag
         wrappedRequest = new HttpServletRequestWrapper((HttpServletRequest) request, bufferedInputStream);
 
-        // OpenTracing - set up scope placeholder.
-        Scope scope = null;
-
-        if (openTracingService.isPresent() && openTracingService.get().isEnabled()) {
-            LOG.trace("We enabled OpenTracing service.  Let's see if there are any passed-in spans");
-
-            SpanContext context = null;
-
-            try {
-                context = openTracingService.get().getGlobalTracer().extract(
-                    Format.Builtin.HTTP_HEADERS, new TracerExtractor(wrappedRequest));
-            } catch (RuntimeException re) {
-                LOG.error("Incoming tracer could not be parsed.  We're going to start a new root span here; even though" +
-                    "this is most likely part of a larger span.  Check out thrown exception for more details", re);
-            }
-
-            LOG.debug("Got the span context from request: {}", context);
-
-            Span activeSpan;
-
-            if (context == null)
-                activeSpan = openTracingService.get().getGlobalTracer()
-                    .buildSpan(String.format("%s %s", wrappedRequest.getMethod(),
-                        wrappedRequest.getRequestURI()))
-                    .withTag(Tags.SPAN_KIND.getKey(), Tags.SPAN_KIND_CLIENT).start();
-            else
-                activeSpan = openTracingService.get().getGlobalTracer()
-                    .buildSpan(String.format("%s %s", wrappedRequest.getMethod(),
-                        wrappedRequest.getRequestURI()))
-                    .asChildOf(context)
-                    .withTag(Tags.SPAN_KIND.getKey(), Tags.SPAN_KIND_CLIENT).start();
-
-            scope = openTracingService.get().getGlobalTracer().scopeManager().activate(activeSpan, false);
+        LOG.trace("Let's see if there are any OpenTracing spans passed-in");
+        SpanContext context = null;
+        try {
+            context = tracer.extract(Format.Builtin.HTTP_HEADERS, new TracerExtractor(wrappedRequest));
+        } catch (RuntimeException re) {
+            LOG.error("Incoming tracer could not be parsed.");
+            LOG.error("We're going to start a new root span here; even though this is most likely part of a larger span.");
+            LOG.error("Check out following exception for more details:", re);
+        }
+        LOG.debug("Got the span context from request: {}", context);
```

Comment:
`context` could still be null. I feel like that makes this log line a bit confusing. Maybe throw this in the `try` after the context is extracted?

Reply:
I updated the verbiage a bit, but I want to log if the context was `null`.